upf/upf_pfcp_api.c

```diff
@@ -147,6 +149,41 @@ init_tp_error_report (pfcp_tp_error_report_t * report,
 
 /* message helpers */
 
+static void
+build_ue_ip_address_information (pfcp_ue_ip_address_pool_information_t **
+				 ue_pool_info)
+{
+  upf_main_t *gtm = &upf_main;
+  upf_nat_pool_t *np;
+  upf_ueip_pool_info_t *ue_p;
+
+  vec_alloc (*ue_pool_info, pool_elts (gtm->ueip_pools));
+
+  pool_foreach (ue_p, gtm->ueip_pools)
+  {
+    pfcp_ue_ip_address_pool_information_t *ueif;
+    vec_add2 (*ue_pool_info, ueif, 1);
+    ueif->ue_ip_address_pool_identity = vec_dup (ue_p->identity);
+    SET_BIT (ueif->grp.fields, UE_IP_ADDRESS_POOL_INFORMATION_POOL_IDENTIFY);
+
+    ueif->network_instance = vec_dup (ue_p->nwi_name);
+    SET_BIT (ueif->grp.fields,
+	     UE_IP_ADDRESS_POOL_INFORMATION_NETWORK_INSTANCE);
+
+    pool_foreach (np, gtm->nat_pools)
+    {
+      if (!(vec_is_equal (np->network_instance, ue_p->nwi_name)))
+	continue;
+      pfcp_bbf_nat_port_block_t *block;
```

Comment:
local variable definition should not follow code

Reply:
done

Reply:
this has not been addressed

Reply:
Sorry, missed this here but fixed for stable branch.